src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,52 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHAR_COUNT = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null"
+                    + "Min char count is "
+                    + MIN_CHAR_COUNT);
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null"
+                    + "Min char count is "
+                    + MIN_CHAR_COUNT);
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null"
+                    + "Min age is "
+                    + MIN_AGE);
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User already exists");
+        }
+        if (user.getLogin().length() < MIN_CHAR_COUNT) {
+            throw new RegistrationException("Login is too short"
+                    + user.getLogin()
+                    + "Min char count is "
+                    + MIN_CHAR_COUNT);
+        }
+        if (user.getPassword().length() < MIN_CHAR_COUNT) {
```

Comment:
You are checking if the user's password length is less than the minimum character count but you are not checking if it's exactly equal to the minimum character count. It's an edge case that should be tested.